server/src/routers/MenusRouter/getAllMenus.js

```diff
@@ -0,0 +1,41 @@
+import getOneMenu from './getOneMenu';
+import locations from '../../config/locations';
+
+/* 
+*   getAllMenus(query)
+*
+*   Summary:
+*       Assemblies queries to the Yale Dining API for all locations and converts the returned data into a list of either Menus objects or MenuItem objects
+*
+*   Parameters:
+*       query - the query object parsed from the request
+*
+*   Return Value:
+*       an array of [MenuItem | Menus] objects, depending on whether the query object contains the meal key or not (respectively)
+*
+*   Throws:
+*       when an exception is caught for a critical threshold of requests to getOneMenu()
+*/
+export default async function getAllMenus(query) {
+    var allMenus = [];
+    const maxErrors = Object.keys(locations).length;
+    var nErrors = 0;
+    for (var location in locations) {
+        try {
+            query.location = location;
+            const menus = await getOneMenu(query);
+            menus.forEach((menu) => allMenus.push(menu));
+        }
+        catch (e) {
+            nErrors++;
+            console.warn(e.message + ' (total: ' + nErrors + ')');
+            if (nErrors >= maxErrors) {
+                throw new Error('Empty object returned for all locations');
+            }
+        }
+    }
+    if ('meal' in query)
```

Comment:
What exactly is happening here again?

Reply:
> What exactly is happening here again?

all this does is make a list of ever college's menu right?

Reply:
Ah yes, I added a comment here on a later version, I think? When the `meal` query is specified, a list of `MenuItem` objects (name and ID) is returned. Since there's a lot of redundancy across dining halls, this line filters out duplicates.